src/self-managed-debug/src/k8s_resource_dumper.rs

```diff
@@ -0,0 +1,303 @@
+        .dump()
+        .await;
+    K8sResourceDumper::<NetworkPolicy>::namespaced(context, client.clone(), namespace.clone())
+        .dump()
+        .await;
+    K8sResourceDumper::<Event>::namespaced(context, client.clone(), namespace.clone())
+        .dump()
+        .await;
+
+    dump_k8s_pod_logs(context, client.clone(), &namespace).await;
+}
+
+/// Write cluster-level k8s resources to a yaml file per resource.
+pub async fn dump_cluster_resources(context: &Context, client: &Client) {
+    K8sResourceDumper::<Node>::cluster(context, client.clone())
+        .dump()
+        .await;
+}
+
+/// Runs `kubectl describe` for a given resource type K and writes the output to a file.
+async fn dump_kubectl_describe<K>(
+    context: &Context,
+    namespace: Option<&String>,
+) -> Result<(), anyhow::Error>
+where
+    K: ListableResource,
+{
+    let resource_type = K::URL_PATH_SEGMENT;
+    let mut args = vec!["describe", &resource_type];
+    if let Some(namespace) = namespace {
+        args.extend(["-n", namespace]);
+    }
+
+    if let Some(k8s_context) = &context.args.k8s_context {
+        args.extend(["--context", k8s_context]);
+    }
+
+    let output = tokio::process::Command::new("kubectl")
+        .args(args)
+        .stderr(std::process::Stdio::null()) // Silence stderr
+        .output()
+        .await?;
+
+    if !output.status.success() {
+        return Err(anyhow::anyhow!(
+            "{}",
+            String::from_utf8_lossy(&output.stderr)
+        ));
+    }
+
+    if output.stdout.is_empty() {
+        let mut err_msg = format!("Describe: No {} found", resource_type);
+        if let Some(namespace) = namespace {
+            err_msg = format!("{} for namespace {}", err_msg, namespace);
+        }
+        eprintln!("{}", err_msg);
+        return Ok(());
+    }
+
+    let file_path = format_resource_path(context.start_time, resource_type, namespace);
+    let file_name = file_path.join("describe.txt");
+    create_dir_all(&file_path)?;
+    let mut file = File::create(&file_name)?;
+    file.write_all(&output.stdout)?;
+
+    println!("Exported {}", file_name.display());
+
+    Ok(())
+}
+
+/// Spawns a new task to run `kubectl describe` for a given resource type K and writes the output to a file.
+pub fn spawn_dump_kubectl_describe_process<K>(
+    context: Context,
+    namespace: Option<String>,
+) -> JoinHandle<()>
+where
+    K: ListableResource,
+{
+    mz_ore::task::spawn(|| "dump-kubectl-describe", async move {
+        if let Err(e) = dump_kubectl_describe::<K>(&context, namespace.as_ref()).await {
+            eprintln!(
+                "Failed to dump kubectl describe for {}: {}",
+                K::URL_PATH_SEGMENT,
+                e
+            );
+        }
+    })
+}
+
+fn format_resource_path(
+    date_time: DateTime<Utc>,
+    resource_type: &str,
+    namespace: Option<&String>,
+) -> PathBuf {
+    let mut path = PathBuf::from(format!(
+        "mz-debug/{}/{}",
```

Comment:
What OS do we want to support? This will only work on Linux/Mac/etc., but might fail on Windows where the path separator is \. If we'd like to be portable, we should use https://doc.rust-lang.org/std/path/constant.MAIN_SEPARATOR.html instead.